ts-sdk/whirlpool/src/token.ts

```diff
@@ -280,8 +311,46 @@ export function getCurrentTransferFee(
   };
 }
 
+/**
+ * Builds the required account extensions for a given mint.
+ *
+ * https://github.com/solana-labs/solana-program-library/blob/3844bfac50990c1aa4dfb30f244f8c13178fc3fa/token/program-2022/src/extension/mod.rs#L1276
+ *
+ * @param {Mint} mint - The mint account to build extensions for.
+ * @returns {ExtensionArgs[]} An array of extension arguments.
+ */
+export function getAccountExtensions(mint: Mint): ExtensionArgs[] {
```

Comment:
nit: I think this function works well for token vault (non ATA). so it is not an issue for now.
If this is used for ATA, `ImmutableOwner` will use a bit space ?

Reply:
Yep but since ATAs are initialized through the ATA program we don't have to worry about reserving space and calculating rent. I'll add a comment to this function though